pkg/schemaconv/openapi_test.go

```diff
@@ -0,0 +1,249 @@
+	}
+
+	legacyTypeMap := normalizeTypes(legacyConversion.Types)
+	newTypeMap := normalizeTypes(newConversion.Types)
+
+	// There are minor differences in new API that are semantically equivalent:
+	//	1. old openapi would replace refs to RawExtensoin with "untyped" and leave
+	//		RawExtension with a non-referenced/nonworking definition. New implemenatation
+	//		makes RawExtensoin "untyped", and leaves references to RawExtension.
+	//	2. old openapi would include "separable" relationship with
+	//		arbitrary/deduced maps where the new implementation leaves it unset
+	//		if it is unset by the user.
+	// Revert the changes
+	deducedName := "__untyped_deduced_"
+	untypedName := "__untyped_atomic_"
+	var revertKnownDifferences func(*schema.Atom)
+	revertKnownDifferences = func(typ *schema.Atom) {
+		if typ.Map != nil {
+			for i, f := range typ.Fields {
+				if f.Type.Inlined != (schema.Atom{}) {
+					revertKnownDifferences(&f.Type.Inlined)
+				} else if f.Type.NamedType != nil && *f.Type.NamedType == "io.k8s.apimachinery.pkg.runtime.RawExtension" {
+					// Before all references to rawExtension were replaced with
+					// untyped. In new implementation we preserve the references
+					// and instead change the raw extension type to untyped
+					f.Type.NamedType = &untypedName
+				}
+				typ.Fields[i] = f
+			}
+
+		}
+
+		if typ.Map != nil && typ.Map.ElementType.NamedType != nil {
+			if len(typ.Map.ElementRelationship) == 0 && typ.Scalar != nil && typ.List != nil && *typ.Map.ElementType.NamedType == "__untyped_deduced_" {
+				// In old implementation arbitrary/deduced map would always also
+				// include "separable".
+				// New implementation has some code paths that dont follow that
+				// (separable is default) so always attaach separable to deduced.
+				typ.Map.ElementRelationship = schema.Separable
+			} else if *typ.Map.ElementType.NamedType == "io.k8s.apimachinery.pkg.runtime.RawExtension" {
+				// See other rawextension comment elsewhere.
+				*typ.Map.ElementType.NamedType = untypedName
+			}
+		}
+	}
+
+	for k, v := range newTypeMap {
+		revertKnownDifferences(&v.Atom)
+		newTypeMap[k] = v
+	}
+
+	newTypeMap["io.k8s.apimachinery.pkg.runtime.RawExtension"] = schema.TypeDef{
+		Name: "io.k8s.apimachinery.pkg.runtime.RawExtension",
+		Atom: schema.Atom{
+			Map: &schema.Map{
+				ElementType: schema.TypeRef{
+					NamedType: &deducedName,
+				},
+			},
+		},
+	}
+
+	legacyYaml, err := schemaToYaml(legacyTypeMap)
+	if err != nil {
+		return err
+	}
+
+	newYaml, err := schemaToYaml(newTypeMap)
+	if err != nil {
+		return err
+	}
+
+	if !reflect.DeepEqual(legacyYaml, newYaml) {
+		return errors.New(cmp.Diff(legacyYaml, newYaml))
+	}
+
+	return nil
+}
+
+func TestCRDOpenAPIConversion(t *testing.T) {
+	files, err := crdFS.ReadDir("testdata/crds")
+	require.NoError(t, err)
+	for _, entry := range files {
+		t.Run(entry.Name(), func(t *testing.T) {
+			t.Parallel()
+			contents, err := crdFS.ReadFile("testdata/crds/" + entry.Name())
```

Comment:
I don't know if that's the test I would have expected for CRDs, you're running this against the v2 processed version. But we should compare the `v2 processed version + proto + conversion` against the `v3 + new conversion`, no?